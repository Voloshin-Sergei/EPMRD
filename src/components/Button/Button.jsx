import React from 'react';

import style from './Button.module.scss';

export const Button = ({ text, className, filled }) => (
  <button className={`${style.button} ${className} ${filled ? style.filled : ''}`} type="button">
    {text}
  </button>
);
