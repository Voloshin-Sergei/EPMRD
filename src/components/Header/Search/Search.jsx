import React from 'react';
import style from './Search.module.scss';

const Search = () => (
  <>
    <p className={style.title}>find your movie</p>
    <form className={style.form}>
      <input className={style.field} type="text" placeholder="search move" />
      <div className={style.buttons}>
        <span className={style.text}>search by</span>
        <ul className={style.list}>
          <li className={style.item}>
            <button className={`${style.button} ${style.tag}`} type="button">
              title
            </button>
          </li>
          <li className={style.item}>
            <button className={`${style.button} ${style.tag}`} type="button">
              genre
            </button>
          </li>
        </ul>
        <button className={`${style.button} ${style.search}`} type="submit">
          search
        </button>
      </div>
    </form>
  </>
);

export default Search;
