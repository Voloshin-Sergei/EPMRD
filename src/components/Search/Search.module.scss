@import 'Styles/variables/colors';

.title {
  margin-bottom: 1rem;
  color: $color-primary;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.field {
  width: 100%;
  margin-bottom: 1rem;
  padding: 1rem;
  color: $color-primary;
  font-size: 1.8rem;
  background-color: lighten($color-primary--dark, 25%);
  border: none;

  &::placeholder {
    font-size: 1.8rem;
  }
}

.buttons {
  display: flex;
  align-items: center;
}

.text {
  margin-right: 1rem;
  color: $color-primary;
  font-size: 1.2rem;
  text-transform: uppercase;
}

.list {
  display: flex;
  flex-wrap: wrap;
  list-style: none;
}

.item {
  margin-right: 1rem;
}

.tag {
  background-color: lighten($color-primary--dark, 40%);

  &:hover {
    background-color: lighten($color-primary--dark, 45%);
  }

  &:focus {
    background-color: $color-secondary;
  }
}

.search {
  margin-left: auto;
  font-size: 1.8rem;
  background-color: $color-secondary;

  &:hover {
    background-color: lighten($color-secondary, 5%);
  }

  &:active {
    transform: scale(0.9);
  }
}
