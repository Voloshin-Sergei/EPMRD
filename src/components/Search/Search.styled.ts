import styled from 'styled-components';
import { colors } from '../../styles/variables/colors.styled';
import { breakpoints } from '../../styles/variables/responsive.styled';
import { fonts } from '../../styles/variables/fonts.styled';

export const StyledContent = styled.div`
  max-width: 1400px;
  margin: 0 auto;
  padding: 2rem 12rem 2rem 12rem;

  @media screen and (max-width: ${breakpoints.mobile}) {
    padding: 2rem;
  }
`;

export const StyledTitle = styled.span`
  display: block;
  margin-bottom: 5rem;
  color: ${colors.colorSecondary};
  font-weight: ${fonts.fontWeightMedium};
  font-size: 3.6rem;
  text-transform: uppercase;

  @media screen and (max-width: ${breakpoints.mobile}) {
    margin-bottom: 4rem;
  }

  @media screen and (max-width: ${breakpoints.mobileSm}) {
    margin-bottom: 2rem;
  }
`;

export const StyledTagLine = styled.span`
  margin-bottom: 1rem;
  color: ${colors.colorPrimary};
  font-size: 2.4rem;
  text-transform: uppercase;
`;

export const StyledForm = styled.form`
  padding: 1rem 0;
`;

export const StyledInput = styled.input`
  width: 100%;
  margin-bottom: 1rem;
  padding: 1rem;
  color: ${colors.colorPrimary};
  font-size: 1.8rem;
  background-color: ${colors.colorPrimaryDarkLighten};
  border: none;

  &::placeholder {
    font-size: 1.8rem;
  
`;

export const StyledButtonsContainer = styled.div`
  display: flex;
  align-items: center;
`;

export const StyledButtonsContainerDesc = styled.span`
  margin-right: 1rem;
  color: ${colors.colorPrimary};
  font-size: 1.2rem;
  text-transform: uppercase;
`;

export const StyledList = styled.ul`
  display: flex;
  flex-wrap: wrap;
  list-style: none;
`;

export const StyledListItem = styled.li`
  margin-right: 1rem;
`;
