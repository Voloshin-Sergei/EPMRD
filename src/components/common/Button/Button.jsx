import React from 'react';
import propTypes from 'prop-types';
import style from './Button.module.scss';

export const Button = ({
  className, variant = 'primary', text, type = 'button',
}) => (
  <button className={`${style.button} ${className} ${style[variant]}`} type={type}>
    {text}
  </button>
);

Button.propTypes = {
  className: propTypes.string.isRequired,
  variant: propTypes.oneOf(['primary', 'secondary']).isRequired,
  text: propTypes.string.isRequired,
  type: propTypes.oneOf(['button', 'submit']).isRequired,
};
