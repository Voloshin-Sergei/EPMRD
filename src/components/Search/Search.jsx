import React from 'react';
import { Button } from 'Components/common/Button';

import style from './Search.module.scss';

const searchTags = ['title', 'genre'];

export const Search = () => (
  <>
    <p className={style.title}>find your movie</p>
    <form className={style.form}>
      <input className={style.field} type="text" placeholder="search movie" />
      <div className={style.buttons}>
        <span className={style.text}>search by</span>
        <ul className={style.list}>
          {searchTags.map((tag, index) => (
            <li key={`${index}_${tag}`} className={style.item}>
              <Button className={style.tag} variant="primary" type="button" text={tag} />
            </li>
          ))}
        </ul>
        <Button className={style.search} text="search" variant="primary" type="button" />
      </div>
    </form>
  </>
);
