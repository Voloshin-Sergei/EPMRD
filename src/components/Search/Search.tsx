import React from 'react';
import { Button } from 'Components/common/Button';
import { useSelector, useDispatch } from 'react-redux';
import { SearchTag } from 'Components/Header/Header';
import { RootState } from 'Store/reducers';
import { setSearchCategory, searchMovieValue, fetchMovies } from 'Store/actions/searchMovieAction';

import style from './Search.module.scss';

export interface SearchProps {
  searchTags: Array<SearchTag>;
}

export const Search: React.FC<SearchProps> = ({
  searchTags,
  onChange,
  onClick,
  activeCategory,
  handleSubmit,
}) => {
  const { searchCategory } = useSelector((state: RootState) => state.searchMovieReducer);

  const dispatch = useDispatch();

  // const moviesSearchCategory = (category: string) => {
  //   console.log(category);
  // };

  // const setSearchValue = (event: React.ChangeEvent<HTMLInputElement>): void => {
  //   dispatch(searchMovieValue(event.target.value));
  // };

  const handleSearchValue = (event: React.ChangeEvent<HTMLInputElement>): void => {
    onChange(event.target.value);
  };

  const handleSearch = () => {
    dispatch(fetchMovies());
  };

  const handleFormSubmit = (event: React.FormEvent): void => {
    event.preventDefault();
    handleSearch();
  };

  return (
    <>
      <p className={style.title}>find your movie</p>
      <form className={style.form} onSubmit={handleFormSubmit}>
        <input
          className={style.field}
          type="text"
          placeholder="search movie"
          onChange={handleSearchValue}
        />
        <div className={style.buttons}>
          <span className={style.text}>search by</span>
          <ul className={style.list}>
            {searchTags.map((tag: SearchTag, index: number) => (
              <li key={`${tag.label}_${index}`} className={style.item}>
                <Button
                  onClick={onClick(tag.type)}
                  className={style.tag}
                  dataTestId="search-tag-btn"
                  activeClassName={tag.type === activeCategory ? style.active : ''}
                >
                  {tag.label}
                </Button>
              </li>
            ))}
          </ul>
          <Button className={style.search} dataTestId="search-btn" onClick={handleSubmit}>
            {'search'}
          </Button>
        </div>
      </form>
    </>
  );
};
