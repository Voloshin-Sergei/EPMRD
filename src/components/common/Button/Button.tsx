import React, { useCallback } from 'react';
import { ButtonStyled } from './Button.styled';

export interface ButtonProps {
  className?: string;
  variant?: 'primary' | 'secondary';
  children: React.ReactNode;
  type?: 'button' | 'submit';
  dataTestId?: string;
  onClick?(data?: string): void;
  typeCategory?: string;
  activeCategory?: string;
  active?: boolean;
  typeBtn?: 'searchBtn' | 'sortTagBtn' | 'filterTagBtn' | 'movieInfoBtn';
}

export const Button: React.FC<ButtonProps> = ({
  className = '',
  variant = 'primary',
  children,
  type = 'button',
  dataTestId,
  onClick,
  active,
  typeBtn,
}) => {
  const handleClick = useCallback(() => {
    if (onClick) {
      onClick();
    }
  }, [onClick]);

  return (
    <ButtonStyled
      variant={variant}
      onClick={handleClick}
      type={type}
      data-test-id={dataTestId}
      className={className}
      active={active}
      typeBtn={typeBtn}
    >
      {children}
    </ButtonStyled>
  );
};
