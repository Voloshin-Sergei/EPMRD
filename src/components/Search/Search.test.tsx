import React from 'react';
import { render } from 'enzyme';
import { Search } from './Search';
import { Button } from 'Components/common/Button';


describe('Search component', () => {
  const tagsList = ['one', 'two', 'three'];

  const component = render(<Search searchTags={tagsList}/>);

  it('should render Search component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should contain search button', () => {
    expect(component.find('[data-test-id="search-btn"]')).toHaveLength(1);
  });

  it('should render search tag buttons on given array', () => {

    expect(component.find('[data-test-id="search-tag-btn"]').length).toEqual(tagsList.length);
  });
});
