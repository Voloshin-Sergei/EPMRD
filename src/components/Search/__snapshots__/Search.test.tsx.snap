// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should render Search component 1`] = `
Array [
  <p
    class="title"
  >
    find your movie
  </p>,
  <form
    class="form"
  >
    <input
      class="field"
      placeholder="search movie"
      type="text"
    />
    <div
      class="buttons"
    >
      <span
        class="text"
      >
        search by
      </span>
      <ul
        class="list"
      >
        <li
          class="item"
        >
          <button
            class="button tag primary"
            data-test-id="search-tag-btn"
            type="button"
          >
            one
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="button tag primary"
            data-test-id="search-tag-btn"
            type="button"
          >
            two
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="button tag primary"
            data-test-id="search-tag-btn"
            type="button"
          >
            three
          </button>
        </li>
      </ul>
      <button
        class="button search primary"
        data-test-id="search-btn"
        type="button"
      >
        search
      </button>
    </div>
  </form>,
]
`;
